### PR TITLE
gh-120026: deprecate (soft) Py_HUGE_VAL macro

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-120026

----
📚 Documentation preview 📚: https://cpython-previews--120027.org.readthedocs.build/

### CODE REVIEW

**Comment by skirpichev** [comment]
As mentioned [here](https://github.com/python/cpython/pull/119701#issuecomment-2147157867) I'm not sure that Py_NAN/INFINITY now are really required.

File: Doc/whatsnew/3.14.rst [hunk 1]
```diff
@@ -446,6 +446,10 @@ Deprecated
   :c:macro:`!isfinite` available from :file:`math.h`
   since C99.  (Contributed by Sergey B Kirpichev in :gh:`119613`.)
 
+* Macro :c:macro:`!Py_HUGE_VAL` is :term:`soft deprecated`, use instead
+  :c:macro:`!Py_INFINITY`.  (Contributed by Sergey B Kirpichev in
+  :gh:`120026`.)
```
Comment (AA-Turner) [inline Doc/whatsnew/3.14.rst]:
```suggestion
* The :c:macro:`!Py_HUGE_VAL` macro is :term:`soft deprecated`,
  use :c:macro:`!Py_INFINITY` instead.
  (Contributed by Sergey B Kirpichev in :gh:`120026`.)
```

File: Misc/NEWS.d/next/C API/2024-06-04-13-38-44.gh-issue-120026.uhEvJ9.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Macro :c:macro:`!Py_HUGE_VAL` is :term:`soft deprecated`.
```
Comment (AA-Turner) [inline Misc/NEWS.d/next/C API/2024-06-04-13-38-44.gh-issue-120026.uhEvJ9.rst]:
```suggestion
The :c:macro:`!Py_HUGE_VAL` macro is :term:`soft deprecated`.
```

File: Include/pymath.h [hunk 3]
```diff
@@ -49,7 +49,7 @@
 
 /* Py_HUGE_VAL should always be the same as Py_INFINITY.  But historically
  * this was not reliable and Python did not require IEEE floats and C99
- * conformity.  Prefer Py_INFINITY for new code.
+ * conformity.  Soft deprecated since Python 3.14, use Py_INFINITY instead.
```
Comment (AA-Turner) [inline Include/pymath.h]:
```suggestion
 * conformity.  The macro was soft deprecated in Python 3.14, use Py_INFINITY instead.
```

**Review by AA-Turner — COMMENTED** [review]
A few wording suggestions.

A

**Comment by skirpichev** [comment]
CC @vstinner

File: Doc/whatsnew/3.14.rst [hunk 4]
```diff
@@ -700,12 +700,21 @@ Porting to Python 3.14
...
 * Macros :c:macro:`!Py_IS_NAN`, :c:macro:`!Py_IS_INFINITY`
   and :c:macro:`!Py_IS_FINITE` are :term:`soft deprecated`,
   use instead :c:macro:`!isnan`, :c:macro:`!isinf` and
   :c:macro:`!isfinite` available from :file:`math.h`
   since C99.  (Contributed by Sergey B Kirpichev in :gh:`119613`.)
 
+* :func:`!asyncio.iscoroutinefunction` is deprecated
+  and will be removed in Python 3.16,
+  use :func:`inspect.iscoroutinefunction` instead.
+  (Contributed by Jiahao Li and Kumar Aditya in :gh:`122875`.)
```
Comment (vstinner) [inline Doc/whatsnew/3.14.rst]: This change looks unrelated.

Comment (skirpichev) [reply Doc/whatsnew/3.14.rst]: Yep, artefact of some conflict resolution.  Fixed, thanks.

**Review by vstinner — APPROVED** [review]
LGTM
